base/pmap.jl

```diff
@@ -64,9 +58,13 @@ Split a collection into at least `min_batch_count` batches.
 Equivalent to `partition(c, max_batch_size)` when `length(c) >> max_batch_size`.
 """
 function batchsplit(c; min_batch_count=1, max_batch_size=100)
+    if min_batch_count < 1
+        throw(ArgumentError("min_batch_count must be > 0, got $min_batch_count"))
+    end
 
-    @assert min_batch_count > 0
-    @assert max_batch_size > 1
+    if max_batch_size < 1
```

Comment:
My intention was to disallow `max_batch_size=1`.
The rationale is that anyone who is trying to iterate over batches of 1 should probably just iterate over the collection directly. Or, put differently, anyone who sets `max_batch_size=x`, and x ends up being `1` probably has a bug, and almost certainly will get sub-optimal performance.